### PR TITLE
fix: Sketch updates in different planks

### PR DESCRIPTION
**Pull request by richburdon**

### CODE REVIEW

File: packages/apps/plugins/plugin-sketch/src/hooks/adapter.ts [hunk 1]
```diff
@@ -49,55 +52,126 @@ export class AutomergeStoreAdapter implements StoreAdapter {
...
+      const currentHeads = A.getHeads(doc);
+      const diff = A.equals(this._lastHeads, currentHeads) ? [] : A.diff(doc, this._lastHeads ?? [], currentHeads);
       diff.forEach((patch) => {
         // TODO(dmaretskyi): Filter out local updates.
         const relativePath = rebasePath(patch.path, accessor.path);
         if (!relativePath) {
           return;
         }
 
+        log.info('update', { id: this._id, patch });
```
Comment (wittjosiah) [inline packages/apps/plugins/plugin-sketch/src/hooks/adapter.ts:174]: debug?

File: packages/apps/plugins/plugin-sketch/src/hooks/adapter.ts [hunk 2]
```diff
@@ -49,55 +52,126 @@ export class AutomergeStoreAdapter implements StoreAdapter {
...
+    // TODO(burdon): Live mode via transient feeds?
+    //
+    // TODO(burdon): Create throttled wrapper.
+    let timeout: ReturnType<typeof setTimeout>;
+    const mutations = new Map<string, { type: string; record: TLRecord }>();
+    this._subscriptions.push(
+      store.listen(
+        ({ changes }) => {
+          // TODO(burdon): Called multiple times on update.
+          log.info('mod', { id: this._id, changes });
```
Comment (wittjosiah) [inline packages/apps/plugins/plugin-sketch/src/hooks/adapter.ts:109]: remove? debug?

File: packages/apps/plugins/plugin-sketch/src/hooks/adapter.ts [hunk 3]
```diff
@@ -26,10 +26,13 @@ export type TLDrawStoreData = {
   content: Record<string, any>;
 };
 
+// TODO(burdon): Very noisy protocol implementation.
+
 /**
  * Ref: https://github.com/LiangrunDa/tldraw-with-automerge/blob/main/src/App.tsx.
  */
 export class AutomergeStoreAdapter implements StoreAdapter {
+  private readonly _id = ['A', 'B', 'C', 'D'][Math.floor(Math.random() * 4)];
```
Comment (wittjosiah) [inline packages/apps/plugins/plugin-sketch/src/hooks/adapter.ts:35]: just for debugging? remove?

File: packages/apps/plugins/plugin-sketch/src/hooks/adapter.ts [hunk 4]
```diff
@@ -139,6 +212,9 @@ export class AutomergeStoreAdapter implements StoreAdapter {
 
       // Update/remove the records in the store.
       store.mergeRemoteChanges(() => {
+        if (updated.size) {
+          store.put(Array.from(updated).map((id) => decode(contentRecords[id])));
```
Comment (wittjosiah) [inline packages/apps/plugins/plugin-sketch/src/hooks/adapter.ts:216]: a lot of this looks like just moving code around, is this the main fix here?

Comment (richburdon) [reply packages/apps/plugins/plugin-sketch/src/hooks/adapter.ts:216]: i did reorganize some things, but somehow the update method got removed.

**Pull request merged**